doc/rst/language/spec/tuples.rst

```diff
@@ -714,6 +715,286 @@ where a comma-separated list of components is valid.
+
+Light Tuples and Heavy Tuples
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Throughout the next few sections, the terms light tuple and heavy tuple
+are used frequently to describe two different ways that tuples can
+capture elements.
+
+Tuple expressions or tuple arguments with default argument intent are two
+examples of light tuples. They store elements by reference where it makes
+sense to do so. Light tuples may be viewed as analagous to a group of
+routine arguments that each have default argument intent.
+
+Tuple variables or tuple arguments with `in` intent are two examples of
+heavy tuples. They store all elements by value and will make a copy of
+each element. Heavy tuples may be viewed as analagous to a group of
+routine arguments that each have the `in` intent.
+
+In short, some or all of the elements of a light tuple may be references,
+while a heavy tuple will never contain a reference.
+
+.. _Tuple_Expression_Behavior:
+
+Tuple Expression Behavior
+-------------------------
+
+Tuple expressions are a form of light tuple. Like most other light tuples,
+tuple expressions capture each element based on the default argument intent
+of the element's type.
+
+   *Example (tuple-expression-behavior.chpl)*.
+
+More specifically:
+
+-  If the default argument intent of the element's type is a variation of
+   `ref`, then the tuple expression will refer to the element instead of
+   capturing it by value.
+-  Otherwise, the tuple expression will capture the element by value.
+
+Consider the following example:
+
+   .. code-block:: chapel
+
+      record R { var x: int; }
+
+      var a: [0..0] int;
+      var i: int;
+      var r: R;
+
+      // The int `i` is copied when captured, but `a` and `r` are not.
+      test((a, i, r));
+
+      // Modify the globals, then print the tuple.
+      proc test(tup) {
+        a[0] = 1;
+        i = 2;
+        r.x = 3;
+
+        // Outputs (1, 0, (x = 3)).
+        writeln(tup);
+      }
+
+   .. BLOCK-test-chapeloutput
+
+      (1, 0, (x = 3))
+
+The tuple expression ``(a, i, r)`` will capture the array ``a`` and the record
+``r`` by `ref`, but will create a copy of the integer ``i``.
+
+   *Rationale*
+
+   Tuple expressions and other forms of light tuple are designed to act like
+   a lightweight bundle of arguments. They behave similarly to the
+   individual arguments of a routine call.
+
+   It would be prohibitively expensive for some argument types (such as
+   arrays) to be copied by default when passed as an argument to a
+   routine call.
+
+   The same logic applies to tuple expressions. When the default argument
+   intent of a value's type is some form of `ref`, a tuple expression will
+   capture the value by reference in order to avoid a potentially
+   expensive copy operation.
+
+Tuple Variable Behavior
+-----------------------
+
+Tuple variables are a form of heavy tuple. Like other heavy tuples, tuple
+variables will copy elements as though the element is an argument with
+the `in` intent.
```

Comment:
"as though the element is an argument with the `in` intent"

->

"in a manner similar to passing the element to an `in` intent argument"